### ISSUE TITLE
Enable 'dependabot' for Egeria repos

### ISSUE THREAD

**Issue by grahamwallis** [odpi/egeria#1574]
Please see https://dependabot.com/#how-it-works

**Comment by planetf1** [comment]
It certainly looks quite interesting. Also been acquired by github.

We do our scans independently with Nexus IQ - but then have the work to make necessary changes. This could make those changes easier, though the implication will be constantly pushing us to the latest (I presume). Are we happy with that? Probably in master we should be generally, though components with interdependencies like spring/spring-boot/spring-data may be more consternation

**Comment by grahamwallis** [comment]
I agree that using dependabot might introduce pressure to constantly update but we would be at liberty to reject a suggested update. It could create more monitoring work for us - but overall the workload should be less as the PRs would have been prepared automatically.

There is also the issue that we've seen elsewhere, that an 'update' to a dependency can actually introduce a security regression. My guess we would need to continue with our current  build scanning, and certainly should not instantly switch to dependabot - but maybe introduce it experimentally to see what it comes up with?

**Comment by planetf1** [comment]
Its a github provided option now - so it just needs someone with admin permission to enable if we're happy.

I'd like to discuss some security related topics at the next face to face egeria session, so suggest we talk through this as one of the tools to aid compliance

**Comment by planetf1** [comment]
Agreed at face2face 2 Oct that we will deploy dependabot 

I will create an initial config file as per https://dependabot.com/docs/config-file/ and try it against a private repo, and then contact the build team to enable dependabot against the main egeria repo.

**Comment by planetf1** [comment]
FYI: https://github.com/dependabot/feedback/issues/705

**Comment by planetf1** [comment]
For a preview, I've enabled docker/javascript/java against my fork -> https://github.com/planetf1/egeria/pulls

**Comment by planetf1** [comment]
Initial Observations
 * Config is fairly easy for Java/maven
 * For Javascript/.docker EVERY package.json/Dockerfile needs adding manually (once) - so config needs updating if new ones added (see issue raised above)
 * the proposed PRs look decent - and respects our use of properties->dependencymanagement-dependency
 * Docker dependencies on java err to Java 13 - I think not, but there are controls to decline the PR/stop checking
 * The 'LATEST' version classification isn't supported - not really an issue - I just happened to use for security checks. we can let dependabot do it itself
 * We def should do manual review
 * Default limits will restrict dependabot to a handful of changes at once - probably a good idea though I reconfigured my repo so I could get a better idea of the scope of its proposals
 * There are a lot of changes overall
 * it's found a few places where there are still dependencies specced in child module poms (we should have all in top level). Obv won't fix these, but we can create a corresponding PR as we review the dependabot PRs
 * compatability score is a neat feature
 * I don't see any evidence it is directly trying to fix up transitive dependencies - perhaps only once we've overriden them ourselves - but if we see remaining issues in code scans, it may well then manage them (need to explore this more)
 * not seen any javascript recommendations (yet) so unclear how well that works

**Comment by planetf1** [comment]
* Where we decline an auto update we may want to consider what action to take - ie preventing further checks, or if we need to force an old level (gaian/db?) then adding to release notes

**Comment by planetf1** [comment]
* After a check I seemed to end up with 43 PRs (fyi @mandy-chessell )

**Comment by planetf1** [comment]
will explore the config a little more later

**Comment by planetf1** [comment]
egeria call 2019/10/08
 - Reviewed example output
 - discussed workflow
 - It was pointed out that these updates are needed to clean our security/dependency scans

Actions
 - Activate dependabot on main repo
 - Cap at 10 active PRs ( to manage workload)
 - Assign initial owners/reviewers for each open PR (Nigel, Graham, - any other volunteers) - and configure dependabot accordingly
 - Reviewers to decide/review/manage required testing, and update future reviews/labels/assignees using dependabot commands if needed
 - Use dependabot function to set appro

Note: not discussed, but will set check frequency to once per week.

**Comment by cmgrote** [comment]
@planetf1 can you please (request to) enable on the connector repos as well (igc, atlas)?  I'd be fine to have the same config as core for simplicity (or even have it uncapped on the connector repos).

**Comment by planetf1** [comment]
sure @cmgrote will do

**Comment by planetf1** [comment]
Opened https://jira.linuxfoundation.org/servicedesk/customer/portal/2/IT-17871 to get dependabot access to our repos.

I'll create config file(s) in the next day or so

**Comment by planetf1** [comment]
@cmgrote My IT request above is for all our repos. However so far I've only done the PR for the egeria repo to add the config file.  I'll let you decide if you want to wait and see if it works for Egeria, add a similar one now, or I can.... You are added as a reviewer in the egeria PR

**Comment by planetf1** [comment]
Dependabot license: https://dependabot.com/terms/

```
Intellectual Property
As part of providing the Service, Dependabot generates software code contributions to the customer’s repositories. For the avoidance of doubt, Dependabot grants to each customer a non-exclusive, worldwide right or license to perform, display, and use the contributions and any content contained in, accessed by or transmitted through Dependabot to customer’s repositories.
```
cc: @grahamwallis

**Comment by planetf1** [comment]
this is now done
